server/src/test/java/org/elasticsearch/gateway/PersistedClusterStateServiceTests.java

```diff
@@ -688,6 +688,57 @@ public void test_persists_updated_column_oid() throws IOException {
         }
     }
 
+    public void test_persisted_state_has_only_renamed_index_metadata_after_rename() throws IOException {
+        try (NodeEnvironment nodeEnvironment = newNodeEnvironment(createDataPaths())) {
+            final PersistedClusterStateService persistedClusterStateService = newPersistedClusterStateService(nodeEnvironment);
+
+            final String uuid = UUIDs.randomBase64UUID(random());
+            String index = "test";
+            String renamedIndex = "renamed_test";
+            long indexVersion = 1;
+
+
+            try (Writer writer = persistedClusterStateService.createWriter()) {
+                ClusterState clusterState = loadPersistedClusterState(persistedClusterStateService);
+                // Need to write full state first before doing incremental writes.
+                // Imitate CREATE TABLE
+                writer.writeFullStateAndCommit(0, ClusterState.builder(clusterState)
+                    .metadata(Metadata.builder(clusterState.metadata())
+                        .version(clusterState.metadata().version() + 1)
+                        .coordinationMetadata(CoordinationMetadata.builder(clusterState.coordinationMetadata()).term(1).build())
+                        .put(IndexMetadata.builder(index)
+                            .version(indexVersion)
+                            .settings(Settings.builder()
+                                .put(IndexMetadata.INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), 1)
+                                .put(IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), 1)
+                                .put(IndexMetadata.SETTING_INDEX_VERSION_CREATED.getKey(), Version.CURRENT)
+                                .put(IndexMetadata.SETTING_INDEX_UUID, uuid)))).incrementVersion().build()
+                );
+
+
+                // Imitate ALTER TABLE RENAME: it creates a new IndexMetadata with the same params but different name and version.
+                clusterState = loadPersistedClusterState(persistedClusterStateService);
+                writer.writeIncrementalStateAndCommit(0, clusterState, ClusterState.builder(clusterState)
+                    .metadata(Metadata.builder(clusterState.metadata())
+                        .version(clusterState.metadata().version() + 1)
+                        .remove(index)
+                        .put(IndexMetadata.builder(renamedIndex)
+                            .version(indexVersion + 1)
+                            .settings(Settings.builder()
+                                .put(IndexMetadata.INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), 1)
+                                .put(IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), 1)
+                                .put(IndexMetadata.SETTING_INDEX_VERSION_CREATED.getKey(), Version.CURRENT)
```

Comment:
We usually prefer unit tests over integration tests but I'm not sure about this one.

I had to replicate what I saw in debugger and relied too much on implementation details to make this test valid.

Maybe it makes sense to keep an integration test instead?

I confirmed that 5.10 branch on state including https://github.com/crate/crate/commit/4a82981501619780ce1156aa5015a627de5ff1e1 fails without the fix and works with it.

On master it works due to https://github.com/crate/crate/commit/a6d58586a2db9f15a6dd96c84a0afdd849c37796. 

We update global metadata on incremental write now, 
because we take tables (metadata.schemas) into account in `Metadata.isGlobalStateEquals` and before we didn't use indices in equality check.

Reply:
@mfussenegger let's remove this and keep an itest?